### PR TITLE
Budget App RoR Capstone Project

### PR DESCRIPTION
**Pull request by VelzckC0D3**
# ✨ Requested Implementations ✨

### ✔️ _Splash Screen_
- _A simple page with the name of the app (WiseWallet) and links to sign in and log in pages._

### ✔️ _Sign up & Log in pages_
- _The user is able to register in the app with full name, email & password `Extra: the user can pick a profile avatar`_
- _The user can log in inside the app with those credentials_
- _If the user is not logged in, they can't access pages that require the user to be logged in_

### ✔️ _Home Page (Categories Page)_
- _When the user is logged in, they are presented with their categories index page_
- _For each category, the user can see their name, icon, & total amount of all the transactions for that specific group_
- _There is a button to add a new category at the bottom that brings the user to the form to create that category_
- _When the user clicks or taps the category, they're redirected to the show page of that category_
- _`Extra: if the user doesn't have any group they'll be presented with a different interface, also they can add movements from here where they're going to be asked to which group belongs that transaction`_

### ✔️ _Transactions Page_
- _For a given category, the list of transactions is presented & ordered by most recent_
- _At the top of the page, the user can see the total amount for that category which is the sum of all the transactions_
- _There is a button to add a new transaction at the bottom that brings the user to the form to create a new transaction_
- _When the user clicks the back button (back) it will bring him to the anterior page_

### ✔️ _"Add New Category" Page_
- _The user fills out the form to create a new category, indicating the name and being able to select a group icon_
- _When the user clicks or taps on add category is redirected to the index page_
- _When the user clicks the back button (back) it will bring him to the anterior page_

### ✔️ _"Add New Transaction" Page_
- _The user fills out a form to create a new transaction with name, amount, & categories (at least one)_
- _When the user clicks or taps add category is redirected to the correct page  `Extra: this depends of the starting point of the form, if it's from a specific group, It'll be automatically added to that group, and if it's from the index, they'll be presented with the option to pick the group which it belongs`_
- _When the user clicks back button (back) it will bring him to the anterior page_

### ✔️ _Testing Requirements_
- _Unit & Integral testing with Rspec & Capybara are setted up in the project_
Screenshot 2023-08-12 093936

### ✔️ _Technical Requirements_
- _Used Postgres as main DB_
- _Used Devise for authentication_
- _All models are validated_
- _Used .erb engine_
- _The project is deployed on render, functional, & accessible through [this link](https://rails-eb2s.onrender.com/)_
- _The database schema is presented like so_
erd_diagram

### 🎈 _Additional Requirements_
- _Included UX improvements, like animations and transitions_
- _Decent desktop design_
- _Side menu to improve the navigability of the app_

### 🔧 _Walkthrough Video_
- _Here is a short video demonstrating each of the implemented functions_

https://github.com/VelzckC0D3/Budget_App/assets/92229666/be96f036-bf63-452c-a0a8-8175f57d9629

# 💡 Thank you for your review! 💡

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1 +1,194 @@
...
+
+<!-- ACKNOWLEDGEMENTS -->
+
+## 🙏 Acknowledgments <a name="acknowledgements"></a>
+
+I would like to thanks my Microverse Team and partners for helping me to get this done.
+
+<p align="right">(<a href="#readme-top">back to top</a>)</p>
+
+<!-- LICENSE -->
```
Comment (mikethreels) [inline README.md]: - [ ] according to the requirements The README should contain an "Acknowledgements" section or similar that gives appropriate credit to [Gregoire Vella on Behance](https://www.behance.net/gregoirevella) the [author of the original design](https://www.behance.net/gallery/19759151/Snapscan-iOs-design-and-branding?tracking_source=), as required by the [Creative Commons license of the design](https://creativecommons.org/licenses/by-nc/4.0/).

**Review by mikethreels — CHANGES_REQUESTED** [review]
Hi Alejandro, 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!

## Highlights
- this is one of the best and smoothest projects I've seen so far, very good job, unfortunately, you just missed 1 small detail in the readme.

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: README.md (hunk 1 above)
Comment (VelzckC0D3) [reply README.md]: @mikethreels oh my... i didn't notice, thank you so much 😅

**Review by cynthiainga — APPROVED** [review]
Hi @VelzckC0D3, 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 



## Highlights

- [ ] Good design ✔️ 
- [ ] Good project 👍🏾 👍🏾 
- [ ] No linters errors ✔️ 

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you take them into account as they can make your code better. Some of them were missed by the previous reviewer and addressing them will really improve your application._

Cheers, and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think the code review was unfair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**